### ISSUE TITLE
4.48.1 breaks sliding window in eager attention for qwen2

### ISSUE THREAD

**Issue by wizyoung** [huggingface/transformers#35924]
### System Info

- `transformers` version: 4.48.1
- Platform: Linux-5.10.0-2.0.0.2-x86_64-with-glibc2.27
- Python version: 3.9.19
- Huggingface_hub version: 0.27.1
- Safetensors version: 0.4.5
- Accelerate version: 0.34.1
- Accelerate config:    not found
- PyTorch version (GPU?): 2.5.1+cu124 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using distributed or parallel set-up in script?: <fill in>
- Using GPU in script?: <fill in>

### Who can help?

@ArthurZucker 

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

None

### Expected behavior

I see v4.48.1 removes the sliding_window mask generation in `_prepare_4d_causal_attention_mask_with_cache_position` but leaves `sliding_window` arguments to `attention_interface`:
```
attn_output, attn_weights = attention_interface(
            self,
            query_states,
            key_states,
            value_states,
            attention_mask,
            dropout=0.0 if not self.training else self.attention_dropout,
            scaling=self.scaling,
            sliding_window=sliding_window,  # main diff with Llama
            **kwargs,
        )
```
This is ok under fa2, but what if using eager attention? Eager attention only accepts attention_mask and does not handle sliding windows. I see this should be a bug?
